app/src/main/java/com/keylesspalace/tusky/components/compose/MediaUploader.kt

```diff
@@ -121,7 +121,12 @@ class MediaUploader @Inject constructor(
                             Log.w(TAG, "empty uri path $uri")
                             throw CouldNotOpenFileException()
                         }
-                        val suffix = path.substring(path.lastIndexOf("."))
+                        val dotIndex = path.lastIndexOf(".")
+                        val suffix = if (dotIndex == -1) {
+                            ".tmp"
+                        } else {
+                            path.substring(dotIndex)
+                        }
```

Comment:
This will probably not do the right thing for a path like `/sdcard/foo.bar/nosuffix`.

Something like the following is probably safer:
```kotlin
val file = File(path)
val suffix = "." + file.extension
```